apis/v1/gateway_types.go

```diff
@@ -602,8 +593,27 @@ type ListenerTLSConfig struct {
 	Options map[AnnotationKey]AnnotationValue `json:"options,omitempty"`
 }
 
-// GatewayTLSConfig specifies frontend tls configuration for gateway.
+// GatewayTLSConfig specifies frontend and backend tls configuration for gateway.
 type GatewayTLSConfig struct {
+
+	// Backend describes TLS configuration for gateway when connecting
+	// to backends.
```

Comment:
It would be helpful to say something here about whether it overrides a BackendTLSPolicy.